aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -94,10 +94,92 @@ void erfinv_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t zabs_cuda_wrapper(scalar_t v) {
+  return v;
+}
+
+template<typename T>
+__host__ __device__ static inline T zabs_cuda_wrapper(thrust::complex<T> v) {
```

Comment:
You don't need to handle complex. Clamping complex values based on their absolute values is probably the right way to handle them, but it doesn't follow our current clamp semantics. We'll likely have to introduce a new function or kwarg to clamp.

Reply:
Yes, I've added this before the discussions on complex behaviour happened. I could remove it, but I kept it since complex tensors are disabled in the cpp code anyway. I kept it so that the code is equivalent to the cpu implementation:
https://github.com/pytorch/pytorch/blob/e26631b33332e7409ac799470b3a4b50b4a8bdc4/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L221-L232

I am happy to remove it though too, unless there's some chance it might be useful for the further implementations.

Reply:
No, let's just kill it for clarity.